core/src/main/java/org/infinispan/container/DataContainer.java

```diff
@@ -116,7 +120,10 @@
     * <p/>
     * If a client needs to iterate through a mutable set of mutable cache entries, it should iterate the container
     * itself rather than iterating through the return of entrySet().
-    *
+    * <p>
+    * This set is a read only backed view of the entries underneath. This set will only show non expired entries when
+    * invoked. The size method of the set will count expired entries for the purpose of having a O(1) time cost compared
+    * to O(N) if it is to not count expired entries.
```

Comment:
To be honest I was talking to @tristantarrant mentioning that the API is just all CRAP. I am all for deprecating some of these.